### PR TITLE
Feat/import pem

### PR DESCRIPTION
**Pull request by Cancuuu**
## Import account with .pem file

- Section created for upload .pem file and change emoji and wallet name
- Drag and drop component created
- 

### Type of changes included:

- [x] Components
- [ ] Business Logic
- [ ] Bug fixes
- [ ] Styling
- [ ] Code Refactor

### Screenshots/Gifs:
https://www.loom.com/share/b6d8042d46b241aebee78f2d6b549c56

### Notes:

*Place special notes here*

### Tested on:

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    setOpenEmojis(false);
+    setOpenEmojiSelector(false);
+  };
+
+  useEffect(() => {
+    if (walletName && walletName !== "") {
+      setDisabled(true);
+    } else {
+      setDisabled(false);
+    }
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx:30]: `setDisabled(walletName && walletName !== '')`

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          <FormItem
+            smallLabel
+            label={t('common.name')}
+            className={classes.formItem}
+            component={(
+              <TextInput
+                fullWidth
+                onChange={e => setWalletName(e.target.value)}
+                type="text"
+              // error={}
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx]: remove commented code

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+  const handleRemoveFile = () => {
+    setFile(null);
+    setDisabled(false);
+  }
+
+  useEffect(() => {
+    if (fileNotSupported) {
+      setTimeout(() => {
+        console.log('arranca setTimeOut');
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: remove console.log

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import { Typography } from '@material-ui/core';
+import RedWarningIcon from '@assets/icons/red-warning-icon.svg';
+import { Plug } from "@components";
+
+const DragDropBox = ({ acceptedExtension, setLoading, setDisabled }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  const [dragActive, setDragActive] = useState(false);
+  const [fileNotSupported, setFileNotSupported] = useState(false);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: as a general rule, we usually use ""positive" naming. I'd rename this to `fileSupported`

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  const [file, setFile] = useState(null);
+
+  const handleRemoveFile = () => {
+    setFile(null);
+    setDisabled(false);
+  }
+
+  useEffect(() => {
+    if (fileNotSupported) {
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: and here use `!fileSupported`

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        console.log('arranca setTimeOut');
+        setFileNotSupported(false);
+      }, 2000)
+    }
+  }, [fileNotSupported]);
+
+  // could be a good feature for the future accept more than one extension
+  const checkDesiredFile = (fileObject) => {
+    setLoading(true);
+    setLoading(false);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: ?

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  const [file, setFile] = useState(null);
+
+  const handleRemoveFile = () => {
+    setFile(null);
+    setDisabled(false);
+  }
+
+  useEffect(() => {
+    if (fileNotSupported) {
+      setTimeout(() => {
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: are you removing this timeout somewhere?

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+  }, [fileNotSupported]);
+
+  // could be a good feature for the future accept more than one extension
+  const checkDesiredFile = (fileObject) => {
+    setLoading(true);
+    setLoading(false);
+    const fileExt = fileObject[0]?.name.split('.').pop();
+    if (fileExt === acceptedExtension) {
+      console.log('extension correcta', fileExt);
+      console.log('file extension', fileObject[0]);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: remove logs

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+        setFileNotSupported(false);
+      }, 2000)
+    }
+  }, [fileNotSupported]);
+
+  // could be a good feature for the future accept more than one extension
+  const checkDesiredFile = (fileObject) => {
+    setLoading(true);
+    setLoading(false);
+    const fileExt = fileObject[0]?.name.split('.').pop();
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: can this break if the name is not properly set?

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+    if (fileExt === acceptedExtension) {
+      console.log('extension correcta', fileExt);
+      console.log('file extension', fileObject[0]);
+      setFile(fileObject[0]);
+      setLoading(false);
+      setDisabled(true);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx:44]: ?

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+    } else {
+      setFileNotSupported(true);
+      console.log('extension incorrecta', fileExt);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: remove all console logs

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+  }
+
+  const handleDrag = (e) => {
+    console.log('entra en el handle Drag');
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.type === "dragenter" || e.type === "dragover") {
+      setDragActive(true);
+    } else if (e.type === "dragleave") {
+      setDragActive(false);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: is there any other type? If these 3 are the only ones you can avoid the second `if`

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+    }
+  }
+
+  const handleDrop = (e) => {
+    console.log('entra en el handle Drop');
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: 👀

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(false);
+    if (e.dataTransfer.files && e.dataTransfer.files[0]) {
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: `if(e.?.dataTransfer?.files?.[0])`

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+      checkDesiredFile(e.dataTransfer.files);
+    }
+  };
+
+  const handleChange = (e) => {
+    console.log('entra en el handleChange');
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: 👀

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.target.files && e.target.files[0]) {
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: same as above

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+
+  return (
+    <>
+      <div className={fileNotSupported ? classes.dragDropContainerError : classes.dragDropContainer}>
+        <form
+          onDragEnter={handleDrag}
+          onDragLeave={handleDrag}
+          onDragOver={handleDrag}
+          onDrop={handleDrop}
+          onSubmit={(e) => e.preventDefault()}
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: is this ok?

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+              <img src={RedWarningIcon} alt="" /> {t('importPem.fileNotSupported')}
+            </span>
+          )
+        }
+      </div>
+    </>
+  )
+}
+
+export default DragDropBox
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: add end of file

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+                      <Plug size="small" /> 
+                      <span className={classes.dropItLabel}>
+                        {t('importPem.dropIt')}
+                      </span>
+                    </div>
+                  ) : (
+                    <>
+                      <CloudUpload style={{ color: '#BBBEC2' }} fontSize='large' />
+                      <Typography className={classes.dragDropText} variant='h6'>
+                        {t('importPem.dragAndDrop')}<br />
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: why use `br` here? A margin would def be better

File: source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+                  )
+                }
+              </>
+            )
+          }
+        </form>
+        {
+          fileNotSupported && (
+            <span className={classes.error}>
+              <img src={RedWarningIcon} alt="" /> {t('importPem.fileNotSupported')}
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx]: why embed the img inside the span? 
it'd be better imo to have a div surrounding an img and a span

File: source/views/Extension/Views/ImportWallet/hooks/useSteps.jsx [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+      />,
+      right: rightButton,
+      center: `${t("importPem.walletDetails")}`,
+    },
+  ];
+
+  return steps[step];
+};
+
+export default useSteps;
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/hooks/useSteps.jsx]: beautiful steps hook 👏

File: source/views/Extension/Views/ImportWallet/styles.js [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+    borderRadius: 6,
+    marginBottom: 20,
+  },
+  dragDropContainerError: {
+    border: '2px dashed #E53C3C',
+    boxShadow: '0px 1px 2px rgba(0, 0, 0, 0.05)',
+    height: '142px',
+    width: '370px',
+    borderRadius: 6,
+    marginBottom: 40,
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/styles.js]: there's a lot of repeated styles here, you could abstract them to a common constant like
```js
  const dragDropBase = {
    boxShadow: '0px 1px 2px rgba(0, 0, 0, 0.05)',
    height: '142px',
    width: '370px',
    borderRadius: 6,
  };
  
  ...
  export default makeStyles((theme) => ({
  dragDropContainer: {
    ...dradDropBase,
    border: '2px dashed #D1D5DB',
    marginBottom: 20,
  },
  dragDropContainerError: {
    ...dragDropBase,
    border: '2px dashed #E53C3C',
    marginBottom: 40,
    }

Comment (rocky-fleek) [reply source/views/Extension/Views/ImportWallet/styles.js]: or even just use `dragDropContainer` as the base and overwrite whatever is needed

File: source/views/Extension/Views/ImportWallet/styles.js [hunk 23]
```diff
@@ -0,0 +1,110 @@
...
+  },
+  insideDragDrop: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    flexDirection: 'column',
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/styles.js:28]: you could create a `flex` class to avoid repeating this everywhere

**Review by rocky-fleek — CHANGES_REQUESTED** [review]

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 24]
```diff
@@ -36,13 +30,44 @@ const Step2 = () => {
...
   }, [currentEmoji, walletName]);
 
+  const onEmojiClick = (_event, emojiObject) => {
+    setCurrentEmoji(emojiObject.emoji);
+    setOpenEmojis(false);
+    setOpenEmojiSelector(false);
+  };
+
+  const createImportedAccount = (pemContent) => {
+    console.log(pemContent);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx]: Remove this

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 25]
```diff
@@ -36,13 +30,44 @@ const Step2 = () => {
...
+    sendMessage(
+      {
+        type: HANDLER_TYPES.IMPORT_PEM_ACCOUNT,
+        params: { icon: currentEmoji, name: walletName, pem: pemContent },
+      },
+      (wallet) => {
+        console.log(wallet);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx]: remove this

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 26]
```diff
@@ -36,13 +30,44 @@ const Step2 = () => {
...
+      }
+    );
+    setLoading(false);
+    navigator.navigate("home");
+  };
+
+  const readPemCreateAccount = () => {
+    setLoading(true);
+    setDisabled(true);
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx]: if it's only disabled while loading, use only the loading field.

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 27]
```diff
@@ -36,13 +30,44 @@ const Step2 = () => {
...
 
+  const onEmojiClick = (_event, emojiObject) => {
+    setCurrentEmoji(emojiObject.emoji);
+    setOpenEmojis(false);
+    setOpenEmojiSelector(false);
+  };
+
+  const createImportedAccount = (pemContent) => {
+    console.log(pemContent);
+    sendMessage(
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx:41]: shouldn't you set the loading to true before this?

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 28]
```diff
@@ -36,13 +30,44 @@ const Step2 = () => {
...
+    navigator.navigate("home");
+  };
+
+  const readPemCreateAccount = () => {
+    setLoading(true);
+    setDisabled(true);
+    let fileReader = new FileReader();
+    fileReader.readAsText(userPemFile);
+    fileReader.onloadend = () => {
+      const content = fileReader.result; // we should check what happens if the user pass an empty pem file
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx:59]: has this been checked? Honestly if that's the case, serialization should fail on the BE right @dantefarias ?

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 29]
```diff
@@ -53,41 +78,44 @@ const Step2 = () => {
...
               />
-            )}
+            }
           />
           <Button
             variant="rainbow"
-            value={t('common.save')}
-            onClick={() => navigator.navigate('home')}
+            value={t("common.save")}
+            onClick={() => readPemCreateAccount()}
```
Comment (rocky-fleek) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx]: ` onClick={readPemCreateAccount}`

**Review by rocky-fleek — CHANGES_REQUESTED** [review]

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx (hunk 28 above)
Comment (dantefarias) [reply source/views/Extension/Views/ImportWallet/Steps/Step2.jsx:59]: If the pem file is invalid or empty BE is gonna throw an error about **Invalid Key**

**Review by rocky-fleek — APPROVED** [review]
LGTM! 🚀

**Pull request merged**